### DISCUSSION TITLE
prefix id for some connections don't seem to work

### DISCUSSION THREAD

**Discussion by nengoxx** [open-webui/open-webui#7999] [Issues]
# Bug Report

## Installation Method

pip

## Environment

- **Open WebUI Version:** 0.4.7

- **Operating System:** Windows 11
- **Browser (if applicable):** Firefox

**Confirmation:**

- [x] I have read and followed all the instructions provided in the README.md.
- [ ] I am on the latest version of both Open WebUI and Ollama.
- [ ] I have included the browser console logs.
- [ ] I have included the Docker container logs.
- [x] I have provided the exact steps to reproduce the bug in the "Steps to Reproduce" section below.

## Expected Behavior:

The prefix should be prepended to the model names from that connection.

## Actual Behavior:

Doesn't seem to work on some connections like Huggingface's serverless API and OpenRouter if you set the specific model IDs.

## Description

**Bug Summary:**
I set up different prefixes for several of my connections to the APIs, but only groq models seemed to actually use the prefix.

## Reproduction Details

**Steps to Reproduce:**
Set a prefix for either huggingface's serverless API or OpenRouter's with specific model IDs.

## Logs and Screenshots

**Screenshots/Screen Recordings (if applicable):**

![Screenshot 2024-12-21 180250](https://github.com/user-attachments/assets/411bba71-e087-4e28-908b-25fa5d67aa1b)
![Screenshot 2024-12-21 180313](https://github.com/user-attachments/assets/d67e2b9e-0b3d-4ed8-a946-9bcba46d0a9c)
![Screenshot 2024-12-21 180359](https://github.com/user-attachments/assets/1e101fb1-2125-40e3-8ca4-edc64490abc5)

## Additional Information

In the last screenshot, one of the gemma2-27b-it is from huggingface ('hf' as prefix) and the other from openrouter ('or' as prefix), groq works as you can see. I tried with different prefixes with no luck.

**Comment by nengoxx** [comment]
To clarify, this is because I manually added the model names for OpenRouter & Huggingface, not sure if it's intended:
![Screenshot 2024-12-22 172437](https://github.com/user-attachments/assets/0f0da230-de30-4b5a-a3ed-585123139397)

**Comment by jwsky** [comment]
me too, same issue.
set https://openrouter.ai/v1 as a server and  xxxx as a prefix, but doesn't work.  Works for openai and other service providers
openai response 
<img width="380" alt="image" src="https://github.com/user-attachments/assets/af3d14d0-9a11-4cc5-86bb-955b6c8b2428" />

openrouter response: seems name already specified
<img width="466" alt="image" src="https://github.com/user-attachments/assets/9f61dfc8-701a-4af7-a6d4-3a964d10794f" />

**Comment by openelearning** [comment]
Hello, same problem here.
The prefix ID works for DeepSeek models and Groq, but it doesn't work for OpenRouter.ai. 
The prefix ID didn't work for Groq the first time (and I thought for Deepseek but I'm not sure) but after made some changes (refresh browser, editing the ID...) it works. But not for OpenRouter.ai.
OpenWeb UI was installed with pip (not Docker) and is updated to version 0.5.18. Models from OpenRouter.ai work fine, but the prefix ID is not displayed. Debian 12 used and firefox.

**Reply by openelearning** [reply]
Version 0.5.20 is also affected, for Alibaba cloud there is no problem, it seems that OpenRouter.ai is the only affected by this bug.

**Comment by silentoplayz** [comment]
See https://github.com/open-webui/open-webui/discussions/9819#discussioncomment-12758548
